react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,307 @@
+            && UIUtil.isButtonEnabled(buttonName);
+
+        dispatch(setToolbarButton(buttonName, {
+            hidden: !isVisible
+        }));
+    };
+}
+
+/**
+ * Shows or hides the dialpad button.
+ *
+ * @param {boolean} show - Flag showing whether to show button or not.
+ * @returns {Function}
+ */
+export function showDialPadButton(show: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'dialpad';
+        const shouldShow = UIUtil.isButtonEnabled(buttonName) && show;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows recording button.
+ *
+ * @returns {Function}
+ */
+export function showRecordingButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const eventEmitter = APP.UI.eventEmitter;
+        const buttonName = 'recording';
+
+        dispatch(setToolbarButton(buttonName, {
+            hidden: false
+        }));
+
+        Recording.init(eventEmitter, config.recordingType);
+    };
+}
+
+/**
+ * Shows or hides the 'shared video' button.
+ *
+ * @returns {Function}
+ */
+export function showSharedVideoButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'sharedvideo';
+        const shouldShow
+            = UIUtil.isButtonEnabled(buttonName)
+                && !config.disableThirdPartyRequests;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows SIP call button if it's required and appropriate
+ * flag is passed.
+ *
+ * @param {boolean} show - Flag showing whether to show button or not.
+ * @returns {Function}
+ */
+export function showSIPCallButton(show: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonName = 'sip';
+        const shouldShow
+            = APP.conference.sipGatewayEnabled()
+                && UIUtil.isButtonEnabled(buttonName)
+                && show;
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonName, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows the toolbar for specified timeout.
+ *
+ * @param {number} timeout - Timeout for showing the toolbar.
+ * @returns {Function}
+ */
+export function showToolbar(timeout: number = toolbarTimeout): Object {
+    return (dispatch: Dispatch<*>, getState: Function) => {
+        const state = getState();
+        const { visible } = state['features/toolbar'];
+        let updateTimeout = false;
+
+        if (interfaceConfig.filmStripOnly) {
+            return;
+        }
+
+        if (!visible) {
+            dispatch(setToolbarVisibility(true));
+            dispatch(setSubjectSlideIn(true));
+
+            updateTimeout = true;
+        }
+
+        if (updateTimeout) {
```

Comment:
The variable `updateTimeout` seems superficial to me. The condition of `if (updateTimeout) {` will be true only if the condition of `if (!visible) {` is true. So why are there 2 separate `if`s instead of 1 then?